src/MELT/TestSink.cs

```diff
@@ -30,14 +30,27 @@ public class TestSink : ITestSink
 
         public Func<BeginScopeContext, bool>? BeginEnabled { get; set; }
 
+
         public IProducerConsumerCollection<BeginScopeContext> BeginScopes { get => _beginScopes; }
 
         public IProducerConsumerCollection<WriteContext> Writes { get => _writes; }
 
         public IEnumerable<LogEntry> LogEntries => Writes.Select(x => new LogEntry(x));
 
+        /// <summary>
+        /// All scopes that were created
+        /// </summary>
         public IEnumerable<BeginScope> Scopes => BeginScopes.Select(x => new BeginScope(x));
 
+        /// <summary>
+        /// All the current scopes.
+        /// </summary>
+        public IEnumerable<BeginScope> CurrentScopeData => _currentScope.ScopeData;
+
+        private (IReadOnlyCollection<BeginScope> ScopeData, IReadOnlyCollection<TestScope> ActiveScopes) _currentScope = (new List<BeginScope>(), new List<TestScope>());
```

Comment:
not a fan of using a tuple here, also I don't really get why we need two different lists, is not the list of active ones what we need?
btw It doesn't really matter year but IEnumerable would be enough  (e.g. not using Count)

Reply:
I'm afraid I think the naming has made things a bit confusing. 

`BeginScope` contains all the data that was added when the scope was opened. It is immutable and will be in the `FullScope` collection of every `LogEntry` created within that scope.

`TestScope` is the `IDisposable` that is returned by the `ILogger.BeginScope()` method. It doesn't contain it's own scope data but it has a reference to the previous state so that we can restore that state once it is disposed.

Perhaps if we renamed `BeginScope` to something like `Scope` or `ScopeData` and renamed `TestScope` to something like `ScopeCallback` it would be more obvious why we need to keep track of both of them.

Reply:
> `BeginScope` contains all the data that was added when the scope was opened. It is immutable and will be in the `FullScope` collection of every `LogEntry` created within that scope.

This is already returned by `Scopes`, so not sure why we also need to store it in _currentScope.ScopeData